### ISSUE TITLE
Reconsider headerbar buttons layout

### ISSUE THREAD

**Issue by nana-4** [themix-project/oomox#185]
Since the current headerbar is packed with many buttons, it has several issues:

- It looks crowded.
- It requires a wide window width.
- Icon buttons are hard to understand their roles until showing the tooltips (since these icons are not very common) - which reduces the usability.

**Comment by nana-4** [comment]
Here's my proposal:

![image](https://user-images.githubusercontent.com/9556384/50387755-7778f900-0747-11e9-9d9e-8bca872da5af.png)


Pros:

- Much looks clean.
- No need to rely on tooltips.

Cons:

- 2 clicks required to push the (menu) buttons.

**Comment by actionless** [comment]
what about having export as a 3 linked buttons:

  ____________  ___________   ___
(  Export theme  |  Export icons  |  ▼  )
                                ___________/\________
                                |                                       |
                                |    Export Terminal          |
                                |    Apply Spotify Theme  |
                                |___________________|

**Comment by actionless** [comment]
also, one more idea to reduce top panel free space -- what about linking not only export but also making all the left-side icons except for Import to be  a big .linked block?

i saw similar pattern in gtk3-widget-factory:
![2019-01-11--1547233431_480x203_scrot](https://user-images.githubusercontent.com/1655669/51054118-073df480-15dc-11e9-888f-1e3a771f610f.png)

**Comment by actionless** [comment]
what do you think?

![2019-01-11--1547247299_1263x237_scrot](https://user-images.githubusercontent.com/1655669/51063749-9eb33f80-15fc-11e9-9d25-1a0964b8619e.png)

![2019-01-11--1547247366_1229x206_scrot](https://user-images.githubusercontent.com/1655669/51063754-a377f380-15fc-11e9-84ab-b6ba8209a8b7.png)

![2019-01-11--1547247341_1237x227_scrot](https://user-images.githubusercontent.com/1655669/51063745-9bb84f00-15fc-11e9-88bb-fac1ec6f26a7.png)

**Comment by actionless** [comment]
![2019-01-12--1547247681_1290x362_scrot](https://user-images.githubusercontent.com/1655669/51063877-344ecf00-15fd-11e9-8704-c248a601d757.png)

**Comment by actionless** [comment]
![2019-01-12--1547247745_1324x391_scrot](https://user-images.githubusercontent.com/1655669/51063906-5cd6c900-15fd-11e9-9479-f998d01b0a1f.png)

**Comment by nana-4** [comment]
I'm sorry for the late reply.

> also, one more idea to reduce top panel free space -- what about linking not only export but also making all the left-side icons except for Import to be a big .linked block?

Maybe somewhat better, but it will **not** solve the issue I mentioned above:

> Icon buttons are hard to understand their roles until showing the tooltips (since these icons are not very common) - which reduces the usability.

**Comment by actionless** [comment]
> (since these icons are not very common)

that's why i replaced clone to "save as" but feel free to propose other more common icon choices

> linked

i can push it to a separate branch so you can test how it looks for you on non-tiling wm

**Comment by nana-4** [comment]
> but feel free to propose other more common icon choices

I'm proposing the text-style **menu items** (except for Save button) instead of icon buttons.

I barely understand what "Save" icon and "Remove" icon are, but can't understand what "Save as" icon and "Rename" icons are at a glance (unless I show tooltips on them).

> i can push it to a separate branch so you can test how it looks for you on non-tiling wm

Thanks, but I'm already testing [how it looks on GNOME](https://user-images.githubusercontent.com/9556384/51201141-a1849c00-193e-11e9-92ab-cd6932478684.png) :)
My impression is the same as above.

**Comment by actionless** [comment]
yeah, but after looking at the tooltip once you will already know it, while extra clicks you'd have to do every time

so the point is what's for to look pretty if it will cause user to do extra work?


> I'm already testing how it looks on GNOME

i see on your screenshot left button group is not linked yet, so after linking them it will be even more compact, i've pushed to branch `linked-buttons`

**Comment by actionless** [comment]
however in gtk3-widget-factory example `X` button is separated from button group, mb it would make sense to do it the same way

**Comment by actionless** [comment]
and also about "save as"/coppy icon readability -- i think previous was more comprehensible (since it had two documents) while current looks almost the same as "save" and hard to distinguish

**Comment by nana-4** [comment]
> yeah, but after looking at the tooltip once you will already know it, while extra clicks you'd have to do every time
> so the point is what's for to look pretty if it will cause user to do extra work?

In order to know it, users need to familiarize and remember it. The point is not only people who frequently use the app enough to remember it.

In my personal experience, I always at a loss which icon button to press while displaying tooltips, because those icon buttons are _still_ ambiguous and difficult to understand to me. So I'd prefer one extra click rather than displaying tooltips for a few seconds while wondering.

Also, as a mouse user, I'd be glad if there is a right-click menu on the sidebar items which has "Save as", "Rename", and "Remove" items.

**Comment by nana-4** [comment]
> and also about "save as"/coppy icon readability -- i think previous was more comprehensible

If you'll stick to the icon button style, I agree with it. Renaming to "Save as" I suggested was premised on making it a text style.

**Comment by actionless** [comment]
```
 _________________________________________
/ Hey buddy!                              \
|                                         |
| I've heard. You're about to make your   |
| cool theme. I'm here to assist you with |
\ that.                                   /
 -----------------------------------------
                \
                  \
```
...................................................![](https://vignette.wikia.nocookie.net/central/images/c/cb/Clippy.png/revision/latest?cb=20110716185608)

**Comment by actionless** [comment]
but jokes aside, Firefox is also showing such help balloons on first start and after major updates

and also we almost already have our own mr mascot guy, so those could be combined :D

**Comment by nana-4** [comment]
> that could be a way to go, i only don't know how to make it discoverable for the new users what there are some extra actions (like renaming when you click on right side)

I suggested a right click menu as a **secondary** menu. I mean, it's a duplicate of the primary hamburger menu.

I believe the hamburger menu is the first place for users to look for something action, so I don't share the concern of discoverability.

> current workflow of creating new theme (editing factory preset and saving it or choosing factory preset and cloning it) could be done somehow more intuitive, but i really have no clue on this part

Regarding that, I think adding "Create New Preset" item in the popover on the top left would help that. Of course, separately from "Save as" (or "Clone").

```
    ___
   | + |
   __/\___________________________________
  |                                       |
  |  Create New Preset                    |
  |  -----------------------------------  |
  |  Import Themix Profile…               |
  |  Import Colors from Base16 YML File…  |
  |  Import Colors from Image…            |
  |_______________________________________|
```

**Comment by nana-4** [comment]
I think adding a small usage guide in the `README.md` is also helpful.

**Comment by actionless** [comment]
>  "Create New Preset"

i am still not sure what that button supposed to do, since themes don't have so-called "initial state"

usage guides are already in readme, in the almost most bottom section

**Comment by nana-4** [comment]
> i am still not sure what that button supposed to do, since themes don't have so-called "initial state", mb just create a copy of random preset (not preset called "random")

I think it's better to make and load an "initial state" preset than to create a copy of random preset.
The "initial state" preset should be very basic and standard-like. ("oodwaita" is a good example. Or maybe we can just reuse it as an "initial state" preset.)

The user's workflow should be as follows:

1. Push `+` sign button then push the `Create New Preset` item.
2. `Input a new preset name` dialog is opened, input the name, then push "OK" button.
3. A very basic preset is now created.

**Comment by actionless** [comment]
yeah, in ui it will be very easy to do, just bind 'new theme' button to "save oodwaita as..." (or other theme)

**Comment by actionless** [comment]
one more idea:

```
[ + ]     Themix        [ Burger ]    | (sidebar split)                      %ThemeName%                       [ Save ] icon            [ Export theme  | v ]     [ : ]
```

**Comment by actionless** [comment]
also it could be nice idea for hobby project to translate ascii-"art" to gnome-UI mockups :D

**Comment by nana-4** [comment]
> however looking on the mockups i see one important difference:
in the mockups they're using Burger menu for app-level actions, and 3-vertical-dots menu for item-level actions. while in your mockups both are residing under Burger menu (which seems semantically incorrect)

I reread the GNOME's guidelines:

- https://developer.gnome.org/hig/stable/primary-menus.html.en
- https://developer.gnome.org/hig/stable/secondary-menus.html.en
- https://developer.gnome.org/hig/stable/sidebar-lists.html.en
- https://gitlab.gnome.org/GNOME/Initiatives/wikis/App-Menu-Retirement

You're right. Themix should have primary and secondary menus.

**Comment by nana-4** [comment]
For balancing, such a buttons placement may be better:

```
[ + ]  Themix  [ = ]  |  [ Export v ]      %ThemeName%      [ Save ]  [ : ]
```

**Comment by actionless** [comment]
please note what in my proposal Export is linked group of Export Theme and dropdown with other exports, that's important about clickpath for the main usecase
